src/main/java/io/github/oliviercailloux/y2018/assisted_board_games/service/ChessService.java

```diff
@@ -25,50 +24,51 @@
 
 	@Inject
 	private QueryHelper helper;
-	
+
 	@Transactional
 	public List<ChessGameEntity> getAllGames() {
 		return em.createQuery(helper.selectAll(ChessGameEntity.class)).getResultList();
 	}
-	
-	@Transactional
-	public List<ChessStateEntity> getAllStates() {
-		return em.createQuery(helper.selectAll(ChessStateEntity.class)).getResultList();
-	}
-	
+
 	@Transactional
 	public List<ChessMoveEntity> getAllMoves() {
 		return em.createQuery(helper.selectAll(ChessMoveEntity.class)).getResultList();
 	}
-	
-	@Transactional 
+
+	@Transactional
 	public ChessGameEntity getGame(int idGame) {
-		Query q = em.createNamedQuery("ChessGameEntity.find").setParameter("id", idGame); 
+		Query q = em.createNamedQuery("ChessGameEntity.find").setParameter("id", idGame);
 		return (ChessGameEntity) q.getSingleResult();
 	}
-	
+
 	@Transactional
-	public ChessStateEntity getLastState (int idGame) {		
-		Query q = em.createNamedQuery("ChessStateEntity.getLastState").setParameter("gameId", idGame);
-		return (ChessStateEntity) q.getSingleResult();		
+	public int getLastGameId() {
+		Query q = em.createNamedQuery("ChessGameEntity.getLastGameId");
+		return (Integer) q.getSingleResult();
 	}
-	
-	@Transactional 
-	public ChessGameEntity getLastGame() {		
-		Query q = em.createNamedQuery("ChessGameEntity.getLastGame");
-		return (ChessGameEntity) q.getSingleResult();
-	}	
-	
+
 	@Transactional
-	public void persist(ChessGameEntity game) {
-		em.persist(game);
+	public ChessMoveEntity getMove(int idMove) {
+		Query q = em.createNamedQuery("ChessMoveEntity.find").setParameter("id", idMove);
```

Comment:
Warnings Eclipse. Cf. tools.adoc.

Reply:
Non, je voulais dire ici : utiliser `createNamedQuery(String, Class)` pour éviter le cast inélégant.

Reply:
Nous avons enlevé le cast et utilisé createNamedQuery(String, Class).